aeron-archive/src/main/cpp/client/RecordingPos.h

```diff
@@ -29,6 +29,9 @@ namespace aeron { namespace archive { namespace client
  *   0                   1                   2                   3
  *   0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
  *  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *  |                         Archive ID                            |
```

Comment:
It would be better if archive id came at the end to not break backwards compatibility.

Reply:
The key ends with vardata (`sourceInformation`) that's why I added `archiveId` first to avoid having fixed size data after the vardata. But doing so breaks existing clients which expect the key fields to be present at certain offsets.

Reply:
I've moved the `archiveId` to the end of the key for the `RecordingPos` counter. I've also used the same label suffix as for other counters.